CMakeLists.txt

```diff
@@ -92,16 +91,87 @@ set(CHPL_LIBUNWIND_UNIQ_CFG_PATH CACHE INTERNAL "Description")
 set(CHPL_QTHREAD_UNIQ_CFG_PATH CACHE INTERNAL "Description")
 set(CHPL_RE2_UNIQ_CFG_PATH CACHE INTERNAL "Description")
 set(CHPL_PE_CHPL_PKGCONFIG_LIBS CACHE INTERNAL "Description")
-# TODO: SOME VARIABLES THAT AREN'T IN THE PRINTCHPLENV OUTPUT, but are useful
-# set(ENV{CHPL_HOME}) - interacts with printchplenv # need to pass -D cmake values to env
-# CHPL_DEVELOPER - possibly not interacts with printchplenv
-# CHPL_LLVM_GCC_PREFIX - interacts with printchplenv
+
+# Set ENV versions of CHPL_ vars that are not in printchplenv output, in the
+# case when they are set from the cmdline, otherwise get them from the environment
+if (NOT DEFINED ${CHPL_HOME})
+  message(NOTICE "CHPL_HOME not set, using $ENV{CHPL_HOME}")
+  set(CHPL_HOME $ENV{CHPL_HOME} CACHE STRING "Override home directory for Chapel" FORCE)
```

Comment:
I would think it better to leave it unset if it isn't defined yet in cmake. If it's an enviro var, printchplenv will handle it.

Reply:
Hmmm... I think it would be ideal of we just got them from `printchplenv` if they were unset, but these particular flags (`CHPL_HOME`, `CHPL_DEVELOPER`, and `CHPL_LLVM_GCC_PREFIX`) don't get printed out from `printchplenv` with the set of flags we are currently using. It seems that adding the `--cmake` or `--make` format flag omits `CHPL_HOME`, even though `--compiler --internal` does print `CHPL_HOME` at least.

Reply:
`printchplenv` doesn't have the internals for handling `CHPL_DEVELOPER` nor `CHPL_LLVM_GCC_PREFIX`, so I've updated it to work with CHPL_HOME for now and future efforts can address (or not) the other two.